File: M/1274-minimum-swaps-to-make-strings-equal/main.go
package main

import (
	"fmt"
)

func minimumSwap(s1 string, s2 string) int {
	x, y := 0, 0
	for i := 0; i < len(s1); i++ {
		if byte(s1[i]) != byte(s2[i]) {
			if byte(s1[i]) == 'x' {
				x++
			} else {
				y++
			}
		}
	}
	if (x+y)%2 != 0 {
		return -1
	}
	ret := 0
	ret += x / 2
	ret += y / 2
	if x%2 > 0 {
		ret += 2
	}
	return ret
}

func main() {
	fmt.Println(minimumSwap("xx", "yy"))
	fmt.Println(minimumSwap("xy", "yx"))
}

File: M/1274-minimum-swaps-to-make-strings-equal/main.rs
struct Solution {}

impl Solution {
  pub fn minimum_swap(s1: String, s2: String) -> i32 {
    let (x, y): (i32, i32) = s1.chars().zip(s2.chars()).filter(|(b1, b2)| b1 != b2).fold(
      (0, 0),
      |(x, y), (b1, b2)| match (b1, b2) {
        ('x', _) => (x + 1, y),
        ('y', _) => (x, y + 1),
        (_, _) => (x, y),
      },
    );

    // println!("{} {}", x, y);
    if x % 2 != y % 2 {
      return -1;
    }
    x / 2 + y / 2 + (x % 2) * 2
  }
}

fn main() {
  for (x, y) in vec![
    ("xy".to_string(), "yx".to_string()),
    ("xx".to_string(), "yy".to_string()),
    ("xx".to_string(), "xy".to_string()),
    ("xxyxyxyxx".to_string(), "xyxyxxxyx".to_string()),
  ] {
    println!("{}", Solution::minimum_swap(x, y));
  }
}
